prov/verbs/src/verbs_srq.c

```diff
@@ -177,19 +191,27 @@ static struct fi_ops_msg fi_ibv_srq_msg_ops = {
 
 int fi_ibv_srq_close(fid_t fid)
 {
-	struct fi_ibv_srq_ep *ep;
-	int rc;
+	struct fi_ibv_srq_ep *srq_ep;
+	int ret;
+
+	srq_ep = container_of(fid, struct fi_ibv_srq_ep, ep_fid.fid);
 
-	ep = container_of(fid, struct fi_ibv_srq_ep, ep_fid.fid);
+	/* All WCs from Receive CQ belongs to SRQ,
+	 * no need to check EP.
+	 * Assumes that all CQs have already been destroyed */
```

Comment:
This might not be the case. But I think it is safe to assume all the EPs associated with the shared context would have been closed. That means no more completions would arrive in CQ for the recv posted to SRQ.

Reply:
Sure, I agree with you. I'll quote you in the comments to this part of the code :) Thanks.
Unfortunately, we are not able to check status of EPs at this point. Will assume that is right case and the upper layer closes EPs, isn't it?

Reply:
Yes. Ideally, SRQ shouldn't be allowed to close if any EP is still bound to it. A reference counter needs to be added to SRQ to check this.

Reply:
I'm going to add the util_srx and util_stx types. And add the util_ep utilization to the verbs/EP structures. Is it OK?

Reply:
I'm good with expanding the util types.

Reply:
👍

Reply:
Thanks! Will implement this